packages/fiori-freestyle/test/integration/__tests__/app.tests.ts

```diff
@@ -0,0 +1,347 @@
+import { rmdirSync, mkdirSync } from 'fs';
+
+const path = require('path');
+
+const host = 'http://localhost';
+const reportDir = path.join(__dirname, 'screenshots');
+
+interface AppConfig {
+    name: string;
+    appId: string;
+    listSelectorKey: string | undefined;
+    selectorKey: string;
+    listPageTitle: string | RegExp;
+    title: string;
+    url: string;
+}
+
+const checkApp = async (config: AppConfig) => {
+	await page.goto(config.url, {
+		waitUntil: 'load',
+		// Remove the timeout
+		timeout: 0
+	});
+
+    try {
+        const selector = await page.waitForSelector(config.selectorKey, { visible: true, timeout: 40000 });
+        expect(selector).toBeDefined();
+
+			// if (config.listSelectorKey) {
+			//     const listSelector = await page.waitForSelector(config.listSelectorKey, { visible: true, timeout: 40000 });
+			//     expect(listSelector).toBeDefined();
+			// }
+    } catch (e) {
+        await page.screenshot({
+            path: path.join(
+                reportDir,
+                // remove http(s)://
+                `${config.url
+                    .replace(/(^\w+:|^)\/\//, '')
+                    .replace(/\//g, '_')
+                    .replace(/\./g, '_')}_${config.name.replace('', '_')}.png`
+            )
+        });
+        throw e;
+    }
+    await expect(page).toMatchElement(config.selectorKey, {
+        text: config.listPageTitle
+    });
+    await expect(page.title()).resolves.toMatch(config.title);
+};
+
+jest.setTimeout(200000);
```

Comment:
Is this a debugging timeout that can be removed?

Reply:
Probably still needed, but will look to make it lower if possible.